src/rusoto_dynamodbstore.rs

```diff
@@ -920,54 +1005,79 @@ impl super::Backend for Backend {
-                    (item, SubOpState { failure_tx: Some(tx), span })
+                    (
+                        item,
+                        SubOpState {
+                            key: key.into_owned(),
+                            failure_tx: Some(tx),
+                            span,
+                        },
+                    )
                 }
                 AtomicWriteSubOperation::HDel(key, fields) => {
-                    let span = info_span!("h_del", ?key);
+                    let span = info_span!("h_del", ?key, error.msg = Empty, otel.status_code = Empty, otel.kind = "client");
                     let names: HashMap<_, _> = fields
                         .into_iter()
                         .enumerate()
                         .map(|(i, f)| (format!("#n{}", i), encode_field_name(f.as_bytes())))
                         .collect();
                     let mut update = Update::default();
-                    update.key = composite_key(key, NO_SORT_KEY);
+                    update.key = composite_key(&key, NO_SORT_KEY);
                     update.table_name = self.table_name.clone();
                     update.update_expression = format!("REMOVE {}", (0..names.len()).map(|i| format!("#n{}", i)).collect::<Vec<_>>().join(", "));
                     update.expression_attribute_names = Some(names);
                     let mut item = TransactWriteItem::default();
                     item.update = Some(update);
-                    (item, SubOpState { failure_tx: None, span })
+                    (
+                        item,
+                        SubOpState {
+                            key: key.into_owned(),
+                            failure_tx: None,
+                            span,
+                        },
+                    )
                 }
             })
             .unzip();
         tx.transact_items = transact_items;
 
         match self.client.transact_write_items(tx).await {
             Err(RusotoError::Service(rusoto_dynamodb::TransactWriteItemsError::TransactionCanceled { reasons, .. })) => {
-                let mut did_fail_conditional = false;
-                for (i, reason) in reasons.iter().enumerate() {
-                    if let Some(code) = &reason.code {
-                        if code == "ConditionalCheckFailed" {
-                            did_fail_conditional = true;
-                            let state = &states[i];
-                            state.span.record("condition_failure", true);
+                let mut err = None;
+                for (i, reason) in reasons.into_iter().enumerate() {
+                    let Some(code) = reason.code.as_deref() else {
+                        continue;
+                    };
+                    let state = &states[i];
+                    state.span.record("otel.status_code", "ERROR");
+                    state.span.record("error.msg", &code);
+                    println!("code for {i} is {code:?}");
```

Comment:
I guess the print is used for debugging, but it should be removed.

Reply:
Thanks for catching!